src/main/java/com/denizenscript/denizencore/tags/core/ProcedureScriptTagBase.java

```diff
@@ -1,90 +1,90 @@
 package com.denizenscript.denizencore.tags.core;
 
-import com.denizenscript.denizencore.objects.core.DurationTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.*;
 import com.denizenscript.denizencore.scripts.queues.ScriptQueue;
-import com.denizenscript.denizencore.tags.TagRunnable;
-import com.denizenscript.denizencore.objects.core.ListTag;
-import com.denizenscript.denizencore.objects.core.ScriptTag;
 import com.denizenscript.denizencore.scripts.containers.core.ProcedureScriptContainer;
-import com.denizenscript.denizencore.tags.Attribute;
-import com.denizenscript.denizencore.tags.ReplaceableTagEvent;
-import com.denizenscript.denizencore.utilities.CoreUtilities;
 import com.denizenscript.denizencore.utilities.ScriptUtilities;
-import com.denizenscript.denizencore.utilities.debugging.Debug;
 import com.denizenscript.denizencore.tags.TagManager;
+import com.denizenscript.denizencore.utilities.text.StringHolder;
+
+import java.util.Map;
 
 public class ProcedureScriptTagBase {
 
     public ProcedureScriptTagBase() {
-        TagManager.registerTagHandler(new TagRunnable.RootForm() {
-            @Override
-            public void run(ReplaceableTagEvent event) {
-                procedureTag(event);
-            }
-        }, "proc");
-    }
-
-    public void procedureTag(ReplaceableTagEvent event) {
 
         // <--[tag]
         // @attribute <proc[<procedure_script_name>]>
         // @returns ObjectTag
         // @description
         // Returns the 'determine' result of a procedure script.
         // -->
-        if (!event.matches("proc")) {
-            return;
-        }
-
-        Attribute attribute = event.getAttributes();
-        ScriptTag script;
-        String path = null;
-        if (attribute.hasParam()) {
-            if (attribute.getParam().indexOf('.') > 0) {
+        TagManager.registerTagHandler(ObjectTag.class, ElementTag.class, "proc", (attribute, object) -> {

```

Comment:
This should be registered with the param input format, rather than using legacy `getParam()` calls

Reply:
Oh, wait, it is, you just labeled it `object` and didn't use it in all calls

Reply:
please fix the name of the param var